File: src/context/reducers/filter.js
import { UPDATE_FILTER_OPTION, DELETE_FILTER_OPTION } from "../../constants/ActionType/index";
import filterInitialState from '../initialStates/filterInitialState'

const selectedChat = (state, { type, payload }) => {
  switch (type) {
    case UPDATE_FILTER_OPTION:
      return {
        ...state,
        base: payload.base,
        clothes: payload.clothes,
        ears: payload.ears,
        hat: payload.hat,
        mouth: payload.mouth,
        eyes: payload.eyes,
      };

    case DELETE_FILTER_OPTION:
      return filterInitialState

    default:
      return state;
  }
};

export default selectedChat;

File: src/filter.py
import json
rankingJsonFile = open('ranking.json')
metaJsonFile = open('_metadata.json')

 
rankingJsonData = json.load(rankingJsonFile)
metaJsonData = json.load(metaJsonFile)

for i, item in enumerate(rankingJsonData):
    
    attribute = metaJsonData[i]['attributes']
    payload = {
        "background": '',
        "base": '',
        "clothes": '',
        "ears": '',
        "eyes": '',
        "hat": '',
        "mouth": '',
    }
    payload['background'] = attribute[0]['value']
    payload['base'] = attribute[1]['value']
    payload['clothes'] = attribute[2]['value']
    payload['ears'] = attribute[3]['value']
    payload['eyes'] = attribute[4]['value']
    payload['hat'] = attribute[5]['value']
    payload['mouth'] = attribute[6]['value']
    
    item['attributes'] = payload

with open('data.json', 'w') as f:
    json.dump(rankingJsonData, f)
 
rankingJsonFile.close()
metaJsonFile.close()